tests/test_cli.py

```diff
@@ -35,25 +35,28 @@ def __train_model(tmpdir) -> str:
     out, err, exitcode = capture(command)
     print(f"err={err}")
     print(f"out={out}")
-    out_str = out.decode("utf-8")
-    match_out = re.findall(
-        r"M.+\/.+\n.+\=0.+[0-9]{2}\.[0-9]*\n.+\=1.+[0-9]{2}\.[0-9]*\n.+\=2.+[0-9]{2}\.[0-9]*\n",
-        out_str,
-    )
-    assert out_str == match_out[0]
-    assert len(out_str) == len(match_out[0])
-    assert exitcode == 0
-    return model_root
+    return out, err, exitcode, model_root
 
 
 def test_cli_outputs_models_at_specified_model_root(tmpdir):
-    model_root = __train_model(tmpdir)
+    out, _, exit_code, model_root = __train_model(tmpdir)
+
+    assert exit_code == 0
     assert path.exists(path.join(model_root, "model_instances"))
     assert path.exists(path.join(model_root, "ideal_answers"))
+    out_str = out.decode("utf-8")
+    out_str = out_str.split("\n")
+    assert re.search(r"Models are saved at: /.+/model_instances", out_str[0])
+    for i in range(0, 3):
+        assert re.search(
+            f"Accuracy for model={i} is [0-9]+.[0-9]+.",
```

Comment:
the ideal regular expression would be 
```
assert re.search(
            f"Accuracy for model={i} is [0-9]+\.[0-9]+\.",
```
But lint test is failing this one. 
```
Error: ./tests/test_cli.py:52:47: W605 invalid escape sequence '\.'
./tests/test_cli.py:52:55: W605 invalid escape sequence '\.'
```
Seems like it should be re.search(r".."), not re.search(f"..")
```
assert re.search(
            r"Accuracy for model={i} is [0-9]+\.[0-9]+\.",
```
But 

> make test

 is failing for this one. 
@beatthat

Reply:
It's because you're using an `f` string to format the string whereas if you want to be able to use regular exp characters unescaped, you need to use an `r` string.

You can fix various ways, but the simplest is probably stick with `f` string and just escaping the `\` like this:

```python
f"Accuracy for model={i} is [0-9]+\\.[0-9]+\\.",
```